tests/infra-validations.sh

```diff
@@ -35,13 +35,23 @@ infra_region=west/
 infra_generated_dir=fabrikam-generated-deploy
 vg_name="spk-infra-hld-vg"
 generate_pipeline_path="$(pwd)/infra-generation-pipeline.yml"
+generate_pipeline_path_local="$(pwd)/../azure-pipelines/templates/infra-generation-pipeline.yml"
 
 validation_test_yaml="rg_name: <insert value>"
 
 shopt -s expand_aliases
 alias spk=$SPK_LOCATION
 echo "SPK Version: $(spk --version)"
 
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  if ! [ -x "$(command -v gsed)" ]; then
+    echo 'Error: gnu-sed is not installed.' >&2
+    exit 1
+  fi
+  alias sed='gsed'
+  echo 'macOS detected'
+fi
+
```

Comment:
I'm not so sure if logic is working because I am still encountering the following error:

```
sed: illegal option -- r
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
```

@dennisseah, can you confirm?

Reply:
It works for me